crates/nu-parser/src/parse_keywords.rs

```diff
@@ -142,7 +142,7 @@ pub fn parse_for(
             let sig = decl.signature();
 
             // Let's get our block and make sure it has the right signature
-            if let Some(arg) = call.positional.get(2) {
+            if let Some(arg) = call.positional_iter().nth(2) {
```

Comment:
I wonder if we should have a couple more helpers to compact this. Like `positional_nth(..)` and `positional_len()`. Thoughts?

Reply:
Yes, definitely. I like `positional_nth()` and `positional_len()`. `Call` currently has `nth()`, which returns an optional cloned positional argument. That seems to be used in very few places, I think it might be a good idea to just have `positional_nth`, which returns an optional reference and call `cloned` at the specific sites where we need a cloned value and not a reference.

Reply:
I added `positional_nth`, `positional_len` and `named_len`.

I removed the old `Call.nth` (it additionally cloned). It turns out we didn't really need it.

I didn't add a `positional_nth_mut` or similar, because there's only one call site that needs a `&mut` for a specific positional.